### ISSUE TITLE
Input generates NullPointerException

### ISSUE THREAD

**Issue by risicle** [veraPDF/veraPDF-library#1051]
I think this issue should probably be opened against https://github.com/veraPDF/veraPDF-validation but that issue tracker is disabled - so you're getting it here.

I've been fuzzing verapdf's validation-model `1.14.105` using kelinci and AFL and have come across some inputs which generate `NullPointerException` when run against code along the lines of:

```java
public class App 
{
    static PDFAFlavour flavour;
    static
    {
        VeraGreenfieldFoundryProvider.initialise();
        flavour = PDFAFlavour.byFlavourId("1a");
    }

    public static void main(String[] args) throws java.io.IOException, ModelParsingException
    {
        try {
            try (
                ByteArrayInputStream inStream = new ByteArrayInputStream(Files.readAllBytes(new File(args[0]).toPath()));
                PDFAParser toValidate = Foundries.defaultInstance().createParser(inStream, flavour)
            ) {
                PDFAValidator validator = ValidatorFactory.createValidator(flavour, false);
                validator.validate(toValidate);
            }
        } catch (EncryptedPdfException|ValidationException|ModelParsingException e) {
            System.out.println(e);
        }
    }
}
```

I have at least 8 such inputs, and I can't say whether they stem from the same bug or not. I'm posting the first here, but don't know whether I should open separate issues for the others. I'm using openjdk `1.8.0_222`.

[00.pdf](https://github.com/veraPDF/veraPDF-library/files/4114128/00.pdf)

**Comment by carlwilson** [comment]
Hi @risicle and thanks for reporting this. If all of the inputs are `NullPointerExceptions` feel free to add them here please. They may not all share a common cause and technically be different bugs but the causes will be similar.

**Comment by risicle** [comment]
These ones are all very similar, I would be surprised if they weren't the same issue:

[01.pdf](https://github.com/veraPDF/veraPDF-library/files/4136491/01.pdf)
[02.pdf](https://github.com/veraPDF/veraPDF-library/files/4136493/02.pdf)
[03.pdf](https://github.com/veraPDF/veraPDF-library/files/4136498/03.pdf)
[04.pdf](https://github.com/veraPDF/veraPDF-library/files/4136499/04.pdf)
[05.pdf](https://github.com/veraPDF/veraPDF-library/files/4136505/05.pdf)
[06.pdf](https://github.com/veraPDF/veraPDF-library/files/4136507/06.pdf)
[07.pdf](https://github.com/veraPDF/veraPDF-library/files/4136508/07.pdf)
[08.pdf](https://github.com/veraPDF/veraPDF-library/files/4136511/08.pdf)

**Comment by risicle** [comment]
A slightly broader selection here:

[09.pdf](https://github.com/veraPDF/veraPDF-library/files/4136577/09.pdf)
[10.pdf](https://github.com/veraPDF/veraPDF-library/files/4136578/10.pdf)
[11.pdf](https://github.com/veraPDF/veraPDF-library/files/4136579/11.pdf)
[12.pdf](https://github.com/veraPDF/veraPDF-library/files/4136580/12.pdf)
[13.pdf](https://github.com/veraPDF/veraPDF-library/files/4136581/13.pdf)
[14.pdf](https://github.com/veraPDF/veraPDF-library/files/4136582/14.pdf)
[15.pdf](https://github.com/veraPDF/veraPDF-library/files/4136583/15.pdf)
[16.pdf](https://github.com/veraPDF/veraPDF-library/files/4136584/16.pdf)
[17.pdf](https://github.com/veraPDF/veraPDF-library/files/4136585/17.pdf)
[18.pdf](https://github.com/veraPDF/veraPDF-library/files/4136586/18.pdf)
[19.pdf](https://github.com/veraPDF/veraPDF-library/files/4136587/19.pdf)
[20.pdf](https://github.com/veraPDF/veraPDF-library/files/4136588/20.pdf)
[21.pdf](https://github.com/veraPDF/veraPDF-library/files/4136589/21.pdf)
[22.pdf](https://github.com/veraPDF/veraPDF-library/files/4136590/22.pdf)
[23.pdf](https://github.com/veraPDF/veraPDF-library/files/4136591/23.pdf)
[24.pdf](https://github.com/veraPDF/veraPDF-library/files/4136592/24.pdf)
[25.pdf](https://github.com/veraPDF/veraPDF-library/files/4136593/25.pdf)
[26.pdf](https://github.com/veraPDF/veraPDF-library/files/4136594/26.pdf)

**Comment by carlwilson** [comment]
> These ones are all very similar, I would be surprised if they weren't the same issue:
> 
> [01.pdf](https://github.com/veraPDF/veraPDF-library/files/4136491/01.pdf)
> [02.pdf](https://github.com/veraPDF/veraPDF-library/files/4136493/02.pdf)
> [03.pdf](https://github.com/veraPDF/veraPDF-library/files/4136498/03.pdf)
> [04.pdf](https://github.com/veraPDF/veraPDF-library/files/4136499/04.pdf)
> [05.pdf](https://github.com/veraPDF/veraPDF-library/files/4136505/05.pdf)
> [06.pdf](https://github.com/veraPDF/veraPDF-library/files/4136507/06.pdf)
> [07.pdf](https://github.com/veraPDF/veraPDF-library/files/4136508/07.pdf)
> [08.pdf](https://github.com/veraPDF/veraPDF-library/files/4136511/08.pdf)

[00.pdf](https://github.com/veraPDF/veraPDF-library/files/4114128/00.pdf) throws it's `NullPointerException` on a different LOC but looks to have the same / similar root cause.
[08.pdf](https://github.com/veraPDF/veraPDF-library/files/4136511/08.pdf) is a little different:
```bash
  <jobsException in thread "main" java.lang.StringIndexOutOfBoundsException: String index out of range: 0
	at java.lang.String.charAt(String.java:658)
```
Thanks for these.

**Comment by carlwilson** [comment]
OK, I've performed a quick analysis. 
Files 01.pdf -> 07.pdf, 09.pdf, 11.pdf -> 15.pdf, and 17.pdf -> 26.pdf all fail with the same `NullPointerException` and stack trace:
```bash
  <jobsException in thread "main" java.lang.NullPointerException
	at org.verapdf.pd.PDObject.knownKey(PDObject.java:65)
	at org.verapdf.parser.PDFParser.getTrailer(PDFParser.java:540)
	at org.verapdf.parser.PDFParser.getXRefSectionAndTrailer(PDFParser.java:421)
	at org.verapdf.parser.PDFParser.getXRefInfo(PDFParser.java:519)
...
```
Files 00.pdf,  10.pdf also throw a `NullPointerException` but a different stack trace and a warning about a missing stream length:
```bash
  <jobsJan 30, 2020 11:15:54 PM org.verapdf.parser.COSParser checkStreamLength
WARNING: Stream length is missing
Exception in thread "main" java.lang.NullPointerException
	at org.verapdf.parser.XrefStreamParser.parseStreamAndTrailer(XrefStreamParser.java:72)
	at org.verapdf.parser.PDFParser.parseXrefStream(PDFParser.java:485)
	at org.verapdf.parser.PDFParser.getXRefSectionAndTrailer(PDFParser.java:423)
	at org.verapdf.parser.PDFParser.getXRefInfo(PDFParser.java:519)
```
File 08.pdf and 16.pdf both fail with a `StringIndexOutOfBoundsException` and the following stack trace:
```bash
  <jobsException in thread "main" java.lang.StringIndexOutOfBoundsException: String index out of range: 0
	at java.lang.String.charAt(String.java:658)
	at org.verapdf.parser.PDFParser.parseXrefTable(PDFParser.java:462)
	at org.verapdf.parser.PDFParser.getXRefSectionAndTrailer(PDFParser.java:420)
	at org.verapdf.parser.PDFParser.getXRefInfo(PDFParser.java:519)
...
```
I suspect that means three issues effectively. We may triage them out but will probably keep them here for now.

**Comment by bdoubrov** [comment]
Just to note that in all these cases the document is broken and can't be parsed. So, we'll fix this providing more graceful handling of these situations converting them to veraPDF errors and stopping further processing anyway.

**Comment by risicle** [comment]
Absolutely - it's just that if you're running this in production, having to catch NPEs is a bit nasty and also... inconclusive? e.g. verapdf-rest will give you back a 500. It's hard for an app to know which way to take that.

**Comment by risicle** [comment]
This is marked as closed, but half of these file still cause NPEs for me using `validation-model` `1.15.36`. These being: 3-7, 9, 11-13, 19, 21, 23, 26.

Am I missing something?

**Comment by bdoubrov** [comment]
The final fix is available in the latest dev build 1.15.37. Please verify

**Comment by risicle** [comment]
:+1: Verified against all 26, many thanks.
